tests/core/test_callable_utils.py

```diff
@@ -170,10 +176,12 @@ def func_with_extra_param_in_docstring(a: int) -> str:
         return str(a)
 
     # We expect the generated SQL to ignore 'b' since it's not in the function signature
-    sql_body = generate_sql_function_body(
-        func_with_extra_param_in_docstring, "test_catalog", "test_schema"
-    )
+    with pytest.warns(UserWarning) as record:
```

Comment:
```suggestion
    with pytest.warns(UserWarning, match=".") as record:
```

We should always use match when using `pytest.warns`.

Reply:
? Isn't that negated by the reference alias? 

The official docs for pytest show this exact example and also explain when match is to be used (capturing a specific single warning regex match statement), which for code readability, I externalized to parse the recorded warnings. 

https://docs.pytest.org/en/stable/how-to/capture-warnings.html#warns 

I can add the match with a wildcard, but that's a no-op regex match all (which is the same as the default behavior)

Reply:
I'm a bit lost. `with pytest.warns(UserWarning, match="<message>")` doesn't work?

Reply:
created an option match for the multiple capture scenarios. It's definitely a bit cleaner than parametrizing the warnings. 
If it becomes too annoying for changes in the future, we can look into parametrizing the matches.